main.c

```diff
@@ -611,11 +611,12 @@ void *socks5_thread(void *thread_data) {
 		i = (sd >= 0);
 	} else {
 		snprintf(tport, MINIBUF_SIZE, "%d", ntohs(port));
+		char *thost_noport = strdup(thost);
```

Comment:
You can declare after line 406 `char *hostname = NULL` then here you write `hostname = strdup(thost)`

Reply:
Also, after `bailout:` you must add:
```
if (hostname)
    free(hostname);
```

Reply:
Added! Also as Sonar was pointing it out.

Reply:
Sorry, I just realise that since `hostname` is allocated only in the `else` block, it is better to free it at the and of that block, hence removing the 
```
 if (hostname)
    free(hostname);
```
at the end of the function.

This is because if we don't enter the else block `hostname` is not initialised and you cannot free it